src/Core/Repositories/Cosmos/EventRepository.cs

```diff
@@ -0,0 +1,152 @@
+﻿using System.Text.Json;
+using System.Text.Json.Serialization;
+using Bit.Core.Models.Data;
+using Bit.Core.Settings;
+using Bit.Core.Utilities;
+using Bit.Core.Vault.Entities;
+using Microsoft.Azure.Cosmos;
+
+namespace Bit.Core.Repositories.Cosmos;
+
+public class EventRepository : IEventRepository
+{
+    private readonly CosmosClient _client;
+    private readonly Database _database;
+    private readonly Container _container;
+
+    public EventRepository(GlobalSettings globalSettings)
+        : this("TODO")
+    { }
+
+    public EventRepository(string cosmosConnectionString)
+    {
+        var options = new CosmosClientOptions
+        {
+            Serializer = new SystemTextJsonCosmosSerializer(new JsonSerializerOptions
+            {
+                DefaultIgnoreCondition = JsonIgnoreCondition.WhenWritingNull,
+                WriteIndented = false
+            }),
+            // ref: https://learn.microsoft.com/en-us/azure/cosmos-db/nosql/tutorial-dotnet-bulk-import
+            AllowBulkExecution = true
+        };
+        // TODO: Perhaps we want to evaluate moving this to DI as a keyed service singleton in .NET 8
+        _client = new CosmosClient(cosmosConnectionString, options);
+        // TODO: Better naming here? Seems odd
+        _database = _client.GetDatabase("events");
+        _container = _database.GetContainer("events");
```

Comment:
💭 https://learn.microsoft.com/en-us/azure/cosmos-db/resource-model

Could we use containers for organizations? Something else?

Or: https://learn.microsoft.com/en-us/azure/cosmos-db/nosql/model-partition-example

Are events then the container, and the database is something company-level for us?